plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -200,52 +205,30 @@ public void setupLaunchConfiguration(ILaunchConfigurationWorkingCopy workingCopy
     workingCopy.setMappedResources(projects.toArray(new IResource[projects.size()]));
   }
 
-  private void checkAndSetPorts() throws CoreException {
-    checkAndSetPorts(getServer(), new PortProber() {
-      @Override
-      public boolean isPortInUse(int port) {
-        return SocketUtil.isPortInUse(port);
-      }
-    });
-  }
-
-  @VisibleForTesting
-  public interface PortProber {
-    boolean isPortInUse(int port);
-  }
-
+  /**
+   * Check if the configured port (or its default) is in use. Returns the port to be used, or throws
+   * an exception if the port is in use.
+   * 
+   * @param portInUse returns true if the argument is a port that is in use
+   * @return the port value ⊂ [0, 65535]
+   * @throws CoreException if the port is in use
+   */
   @VisibleForTesting
-  void checkAndSetPorts(IServer server, PortProber portProber) throws CoreException {
-    serverPort = checkPortAttribute(server, portProber,
-        SERVER_PORT_ATTRIBUTE_NAME, DEFAULT_SERVER_PORT);
-    adminPort = checkPortAttribute(server, portProber,
-        ADMIN_PORT_ATTRIBUTE_NAME, DEFAULT_ADMIN_PORT);
-  }
-
-  private static int checkPortAttribute(IServer server, PortProber portProber,
-      String attribute, int defaultPort) throws CoreException {
-    int port = server.getAttribute(attribute, defaultPort);
+  static int checkPort(Integer portOrNull, int defaultValue,
```

Comment:
The code is called to validate the settings in a _cloudsdk_'s `RunConfiguration` where `null` is ok (e.g., `RunConfiguration#getApiPort()`).  But doing an `ifNull()` would simplify the test cases.